src/app/auth-modal/login/login.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, OnInit, OnDestroy } from "@angular/core";
+import { FormGroup, FormBuilder, Validators } from "@angular/forms";
+import { of, Subscription } from "rxjs";
+import { catchError, tap } from "rxjs/operators";
+
+import { UserService } from "src/app/core/services/user.service";
+
+import { AuthModalService } from "../auth-modal.service";
+
+@Component({
+  selector: "app-login",
+  templateUrl: "./login.component.html",
+  styleUrls: ["./login.component.less"]
+})
+export class LoginComponent implements OnInit, OnDestroy {
+  public loginForm: FormGroup;
+  public serverError: string | null = null;
+
+  private subs = new Subscription();
+
+  constructor(
+    private fb: FormBuilder,
+    private userService: UserService,
+    private authModalService: AuthModalService
+  ) {
+    this.loginForm = this.fb.group({
+      username: ["", Validators.required],
+      password: ["", Validators.required]
+    });
+  }
+
+  ngOnInit(): void {
+    this.subs.add(
+      this.loginForm.valueChanges.subscribe(() => {
```

Comment:
you can move this to onSubmit error handler and take only first change - you don't need to set this as null every time the form value changes